cdap-api/src/main/java/co/cask/cdap/api/templates/plugins/PluginVersion.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.templates.plugins;
+
+import co.cask.cdap.api.annotation.Beta;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+
+/**
+ * Represents version of a plugin. It will try to parse the version string of format
+ *
+ * <pre>
+ * {@code
+ * parse version string in format of [major].[minor].[fix](-|.)[suffix]
+ * }
+ * </pre>
+ */
+@Beta
+public final class PluginVersion implements Comparable<PluginVersion> {
+
+  private static final Pattern PATTERN = Pattern.compile("(\\d+)(?:\\.(\\d+))?(?:\\.(\\d+))?(?:[.\\-](.*))?");
+
+  private final String version;
+  private final Integer major;
+  private final Integer minor;
+  private final Integer fix;
+  private final String suffix;
+
+  /**
+   * Constructs an instance by parsing the given string.
+   *
+   * @param str the version string. The whole string needs to match with the version pattern supported by this class.
+   */
+  public PluginVersion(String str) {
+    this(str, false);
+  }
+
+  /**
+   * Constructs an instance by parsing the given string.
+   *
+   * @param str the version stirng.
+   * @param matchSuffix if {@code true}, try to match the version pattern by the suffix of the string. Otherwise match
+   *                    the whole string.
+   */
+  public PluginVersion(String str, boolean matchSuffix) {
```

Comment:
Oh, this constructor was used in other places in the plugin system, which is not included in this PR.
Sure, I'll add some test on it in this PR.